certbot-apache/certbot_apache/parser.py

```diff
@@ -188,18 +192,19 @@ def ensure_augeas_state(self):
         """
 
         if self.unsaved_files():
-            self.save_notes += "(autosave)"
+            self.configurator.save_notes += "(autosave)"
             self.configurator.save()
 
-    def save(self):
+    def save(self, save_files=None):
```

Comment:
I think not calling `unsaved_files()` in this function is fine, but if we do that, I think `save_files` should be a required parameter.